compiler/optimizer/RegDepCopyRemoval.hpp

```diff
@@ -105,7 +106,7 @@ class RegDepCopyRemoval : public TR::Optimization
    NodeChoice &getNodeChoice(TR_GlobalRegisterNumber reg);
 
    void discardAllNodeChoices();
-   void discardNodeChoice(TR_GlobalRegisterNumber reg);
+   void discardNodeChoice(TR_GlobalRegisterNumber reg, bool clearRegStore = false);
```

Comment:
I don't like this boolean flag - could we have `discardNodeChoice` and something like `discardNodeChoiceAndClearStore` or something like that? it will be easier to follow the logic rather than having a random bool flag.

Reply:
As the only case when we need to clear up the Node Choice completely is when we start walking new block which is not extension, at that point it cleans up all the data structures that was filled for previous analysis. I renamed this to `clearNodeChoice`.